### ISSUE TITLE
Support export to GitLab snippets

### ISSUE THREAD

**Issue by Aspie96** [hedgedoc/hedgedoc#508]
It would be really nice to have support for GitLab Snippets (instead of GitHub Gists only) to publish pages.
In general, GitLab login would be really nice: it's much more open source than GitHub.

**Comment by davidmehren** [comment]
We already support login with Gitlab, as you can see [in our docs](https://github.com/codimd/server/blob/master/docs/guides/auth/gitlab-self-hosted.md). 

Snippets sound nice, I'll add a feature request label.

**Comment by ErikMichelson** [comment]
GitLab snippet export is indeed an interesting feature.

For GitLab login, see this guide: https://github.com/codimd/server/tree/master/docs%2Fguides%2Fauth%2Fgitlab-self-hosted.md

**Comment by Aspie96** [comment]
Ops, sorry.
I apologize about the login thing, I really didn't notice that!

**Comment by Aspie96** [comment]
I think GitLab support would also be really nice on https://demo.codimd.org/, although not necessary (since e-mail sign up is already possible)

**Comment by storrgie** [comment]
I think that this functionality is currently broken in the 1.6.1 release.

**Comment by davidmehren** [comment]
@storrgie Could you elaborate what exactly is broken in a new issue? We can then keep this issue about the feature request. Thanks!

**Comment by storrgie** [comment]
So we're charting with gitlab-ee (13.5.1-ee right now) and your current images via docker hub. We've got authentication working just fine, however when you go to do an export, the "available projects" won't populate.

This could also be a configuration/settings thing between codimd and gitlab (we're using private groups/projects __everywhere__). To test that we'd not scoped this properly we gave it all permissions aside from sudo and still couldn't get it to list projects.
